tests/test_app.py

```diff
@@ -892,6 +892,173 @@ def testDeleteInsertWhere(self):
+                "o": {'type': 'uri', 'value': 'urn:z'}})
+
+            # test select after
+            obj = json.loads(select_resp_after.data.decode("utf-8"))
+            self.assertEqual(len(obj["results"]["bindings"]), 1)
+            self.assertDictEqual(obj["results"]["bindings"][0], {
+                "g": {'type': 'uri', 'value': 'http://aksw.org/'},
+                "s": {'type': 'uri', 'value': 'urn:x'},
+                "p": {'type': 'uri', 'value': 'urn:1'},
+                "o": {'type': 'literal', 'value': 'new'}})
+
+            # test prov after
+            changesets = json.loads(prov_after.data.decode("utf-8"))
+            self.assertEqual(len(changesets["results"]["bindings"]), 3)
+            self.assertDictEqual(changesets["results"]["bindings"][0], {
+                "op": {'type': 'uri', 'value': 'http://quit.aksw.org/vocab/additions'},
+                "s": {'type': 'uri', 'value': 'urn:x'},
+                "p": {'type': 'uri', 'value': 'urn:y'},
+                "o": {'type': 'uri', 'value': 'urn:z'}})
+            self.assertDictEqual(changesets["results"]["bindings"][1], {
+                "op": {'type': 'uri', 'value': 'http://quit.aksw.org/vocab/additions'},
+                "s": {'type': 'uri', 'value': 'urn:x'},
+                "p": {'type': 'uri', 'value': 'urn:1'},
+                "o": {'type': 'literal', 'value': 'new'}})
+            self.assertDictEqual(changesets["results"]["bindings"][2], {
+                "op": {'type': 'uri', 'value': 'http://quit.aksw.org/vocab/removals'},
+                "s": {'type': 'uri', 'value': 'urn:x'},
+                "p": {'type': 'uri', 'value': 'urn:y'},
+                "o": {'type': 'uri', 'value': 'urn:z'}})
+
+            # compare file content
+            with open(path.join(repo.workdir, 'graph_0.nq'), 'r') as f:
+                self.assertEqual('<urn:x> <urn:1> "new" <http://aksw.org/> .', f.read())
+            with open(path.join(repo.workdir, 'graph_1.nq'), 'r') as f:
+                self.assertEqual('', f.read())
+
+    def testMultioperationalUpdateProvenance(self):
+        """Test multioperational update and compare created provenance information.
+
+        1. Prepare a git repository with an empty and a non empty graph
+        2. Start Quit
+        4. Execute multioperational update query
+        5. Execute SELECT query
+        6. Re-start Quit
+        7. Execute SELECT query
+        8. Compare both query results
+        """
+        # Create querystrings
+        prov = 'SELECT DISTINCT ?op ?s ?p ?o '
+        prov += 'WHERE {?activity <http://quit.aksw.org/vocab/updates> ?update ; '
+        prov += '<http://www.w3.org/ns/prov#endedAtTime> ?time . '
+        prov += '?update ?op ?g . GRAPH ?g {?s ?p ?o .} FILTER ( '
+        prov += 'sameTerm(?op, <http://quit.aksw.org/vocab/additions>) '
+        prov += '|| sameTerm(?op, <http://quit.aksw.org/vocab/removals>) ) } '
+        prov += 'ORDER BY ?time ?update ?g ?s ?p ?o'
```

Comment:
Why don't you just compare the result of `select * { { graph ?g {?s ?p ?o} union {?s ?p ?o}}`?

Reply:
This will lead to different resultsets due to usage of blank nodes in provenance graph.

Reply:
ok